source/agora/consensus/data/Transaction.d

```diff
@@ -247,3 +274,45 @@ public bool getSumOutput (const Transaction tx, ref Amount acc)
             return false;
     return true;
 }
+
+unittest
+{
+    import vibe.data.json;
+    Transaction old_tx = Transaction(
+        TxType.Data,
+        [Input(Hash.init, 0)],
+        [Output.init],
+        [1,2,3]
+    );
+    auto json_str = old_tx.serializeToJsonString();
+
+    Transaction new_tx = deserializeJson!Transaction(json_str);
+    assert(new_tx.data.length == old_tx.data.length);
+    assert(new_tx.data == old_tx.data);
+}
+
+/*******************************************************************************
+
+    Caculates the fee of transaction data to store
+
+    Params:
+        data_size = The size of the data
+        factor = The factor to calculate for the fee of transaction data
+
+    Return:
+        Fees payable to data storage
+
+*******************************************************************************/
+
+public Amount clculateDataFee(ulong data_size, uint factor) @safe nothrow
+{
+    const double decimal = 100.0;
+    ulong fee = cast(ulong)((round((exp(cast(double)data_size / cast(double)factor) - 1.0) * decimal) / decimal) * 10_000_000L);
+    return Amount(fee);
+}
+
+unittest
+{
+    assert(clculateDataFee(0, 200) == Amount(0L));
```

Comment:
Are you going to allow data_size = 0?
Do you have any intentions?

Reply:
When the data size is zero, the fee for data storage is zero.

Reply:
A fee of 0.01 for a typical transaction will be applied once the future development direction is established.